### PR TITLE
Gitlab deploy option

### PR DESCRIPTION
**Pull request by ybizeul**
Added deploy for Gitlab Pages

https://docs.gitlab.com/ee/user/project/pages/

### CODE REVIEW

File: deploy/gitlab.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    if [ -z "$Le_Deploy_gitlab_domain" ]; then
+      _err "GITLAB_DOMAIN not defined."
+      return 1
+    fi
+  else
+    Le_Deploy_gitlab_domain="$GITLAB_DOMAIN"
+    _savedomainconf Le_Deploy_gitlab_domain "$Le_Deploy_gitlab_domain"
+  fi
+
+  curl -s --fail --request PUT --header "PRIVATE-TOKEN: $Le_Deploy_gitlab_token" --form "certificate=@$_cfullchain" --form "key=@$_ckey" "https://gitlab.com/api/v4/projects/$Le_Deploy_gitlab_project_id/pages/domains/$Le_Deploy_gitlab_domain" >/dev/null && exit 0
```
Comment (Neilpang) [inline deploy/gitlab.sh]: do not use `curl`,  please use `_post()`  function instead.

Comment (ybizeul) [reply deploy/gitlab.sh]: Good point, done!

**Comment by Neilpang** [comment]
why is there changes in dns_inwx.sh ?

**Comment by ybizeul** [comment]
It's probably pulled from a more recent dev branch commit when I checked out on another computer. Not sure how to fix this but I'll try

**Comment by ybizeul** [comment]
All good!

File: deploy/gitlab.sh (hunk 1 above)
Comment (zloster) [reply deploy/gitlab.sh]: The _post() function is missing one crucial option: `--fail`. Currently when the GitLab API returns `response='{"message":"401 Unauthorized"}'` the script threats this as success.

Here is an excerpt from my recent attempt:
```
[вт юни 23 21:51:05 EEST 2020] _post_url='https://gitlab.com/api/v4/projects/XXXXXXX/pages/domains/www.edno.moe'
[вт юни 23 21:51:05 EEST 2020] _CURL='curl -L --silent --dump-header /home/XXXXXXX/.acme.sh/http.header  -g '
[вт юни 23 21:51:05 EEST 2020] _ret='0'
[вт юни 23 21:51:05 EEST 2020] response='{"message":"401 Unauthorized"}'
[вт юни 23 21:51:05 EEST 2020] Certificate successfully deployed
```

Comment (ybizeul) [reply deploy/gitlab.sh]: Instead of watching for "error" we could watch for something that is in a successful answer and fail otherwise.

Comment (zloster) [reply deploy/gitlab.sh]: I don't have enough knowledge to be more conclusive.
I've checked the logs. IMO the `"verified":true` (in the response) is a good candidate.

**Pull request merged**